hazelcast/src/test/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorServiceBasicTest.java

```diff
@@ -874,6 +904,49 @@ public void schedule_thenCancelAndGet() throws Exception {
         first.get();
     }
 
+    @Test
+    public void schedule_whenAutoDisposable_thenGet() throws Exception {
+        HazelcastInstance[] instances = createClusterWithCount(2);
+        IScheduledExecutorService executorService = getScheduledExecutor(instances, "s");
+
+        IScheduledFuture<Double> future = executorService.schedule(autoDisposable(new PlainCallableTask()), 1, SECONDS);
+
+        expected.expect(ExecutionException.class);
+        expected.expectCause(isA(StaleTaskException.class));
+        Thread.sleep(2000);
```

Comment:
`Thread.sleep` as you already mentioned is prone to cause flakiness. See `assertTrueEventually` which works with the maximum test waiting time.